torch/csrc/jit/script/convert_to_ssa.cpp

```diff
@@ -223,9 +279,11 @@ struct SSATransformer {
   std::shared_ptr<ValueEnvironment> environment_stack = nullptr;
 };
 
-// Converting to SSA works in two parts. First we add outputs to control flow
+// Converting to SSA works in multiple parts. First we inline the loop condition
+// before and into the body of loops, then we add outputs to control flow
 // nodes, then we stitch together Loads & Stores into SSA form.
 void ConvertToSSA(std::shared_ptr<Graph>& graph) {
+  inlineLoopCondition(graph->block());
```

Comment:
I think this should be a separate pass. It is possible we will want to introduce load/stores in other passes and then re-lower, but putting inlineLoopCondition here prevents it.